---
title: "Untitled"
output: md_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DESeq2)
library(tidyverse)
library(viridis)
library(cowplot)

knitr::opts_chunk$set(fig.path = '../figures/02_RNAseqCA1integrative/')
```

## CA1 setup for deseq

```{r coldata}
# data from other rnaseq study
# fiture to keep only ca1 samples
int.colData <- read.csv("~/GitHub/IntegrativeProjectWT2015/data/02a_colData.csv", check.names = F)
int.countData <- read.csv("~/GitHub/IntegrativeProjectWT2015/data/02a_countData.csv", check.names = F)

int.colData <- int.colData %>%
      dplyr::filter(Punch == "CA1") %>%
      droplevels()

names(int.countData)[1]<-"geneID"
row.names(int.countData) <- int.countData$geneID

savecols <- as.character(int.colData$RNAseqID) 
savecols <- as.vector(savecols) 

int.countData <- int.countData %>% dplyr::select(one_of(savecols)) 


# data from this fmr1 study
fmr.colData <- read.csv("../data/fmr1ColData.csv", header = T)
fmr.countData <- read.csv("../data/fmr1CountData.csv", header = T, check.names = F, row.names = 1)

# confirm proper subsetting
print(ncol(fmr.countData) == nrow(fmr.colData))
print(ncol(int.countData) == nrow(int.colData))

# fill in details to make datasets mergable
int.colData$Genotype <- "WT"
fmr.colData$APA2 <- "yoked_consistent"
fmr.colData$Punch <- "CA1"

fmr.colData$treatment <- paste(fmr.colData$Genotype, fmr.colData$APA, sep = "_")
int.colData$treatment <- paste(int.colData$Genotype, int.colData$APA, sep = "_")

int.colData <- int.colData %>% select(RNAseqID, Mouse, Punch, Genotype, APA2, treatment)
fmr.colData <- fmr.colData %>% select(RNAseqID, Mouse, Punch, Genotype, APA2, treatment)

colData <- rbind(fmr.colData, int.colData)


fmr.countData$geneID <- row.names(fmr.countData)
int.countData$geneID <- row.names(int.countData)

countData <- inner_join(fmr.countData, int.countData)

colData$Punch <- as.factor(colData$Punch)
colData$treatment <- as.factor(colData$treatment)
row.names(countData) <- countData$geneID
countData$geneID <- NULL
row.names(colData) <- colData$RNAseqID
head(colData)
```



```{r}
 # check that row and col lenghts are equal
print(ncol(countData) == nrow(colData))

dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ treatment )
dds <- dds[ rowSums(counts(dds)) > 2, ] ## pre-filter genes 
dds <- DESeq(dds) # Differential expression analysis
vsd <- vst(dds, blind=FALSE) # variance stabilized 

#create list of groups
a <- levels(colData$treatment)
b <- a

numDEGs <- function(group1, group2){
  res <- results(dds, contrast = c("treatment", group1, group2), independentFiltering = T)
  sumpadj <- sum(res$padj < 0.1, na.rm = TRUE)
  return(sumpadj)
}

# comapre all contrasts, save to datafrmes
dat=data.frame()
for (i in a){
  for (j in b){
    if (i != j) {
      k <- paste(i,j, sep = "") #assigns usique rownames
      dat[k,1]<-i               
      dat[k,2]<-j
      dat[k,3]<- numDEGs(i,j) #caluculates number of DEGs
    }
  }
}

head(dat)

# widen data to create table of degs
rownames(dat) <- NULL #remove row names
data_wide <- spread(dat, V2, V3)
print(data_wide) 

# set factors
allcontrasts <- dat %>%
  ggplot( aes(V1, V2)) +
    geom_tile(aes(fill = V3)) +
    scale_fill_viridis(na.value="#FFFFFF00") + 
    xlab(" ") + ylab("Timepoint") +
    labs(fill = "# of DEGs",
         subtitle = "CA1")
plot(allcontrasts)

# create the dataframe using my function pcadataframe
pcadataframe <- function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
  rv <- rowVars(assay(object))
  select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
  pca <- prcomp(t(assay(object)[select, ]))
  percentVar <- pca$sdev^2/sum(pca$sdev^2)
  if (!all(intgroup %in% names(colData(object)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(object)[, intgroup, 
                                               drop = FALSE])
  group <- if (length(intgroup) > 1) {
    factor(apply(intgroup.df, 1, paste, collapse = " : "))
  }
  else {
    colData(object)[[intgroup]]
  }
  d <- data.frame(PC1 = pca$x[, 1], PC2 = pca$x[, 2], PC3 = pca$x[, 3], PC4 = pca$x[, 4],PC5 = pca$x[, 5],PC6 = pca$x[, 6],group = group, 
                  intgroup.df, name = colnames(object))
  if (returnData) {
    attr(d, "percentVar") <- percentVar[1:6]
    return(d)
  }
}


pcadata <- pcadataframe(vsd, intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcadata, "percentVar"))
percentVar

pca12 <- ggplot(pcadata, aes(PC1, PC2,color = treatment)) + 
  geom_point(size = 2, alpha = 1) +
  stat_ellipse(type = "t") +
  xlab(paste0("PC1: ", percentVar[1],"% variance")) +
  ylab(paste0("PC2: ", percentVar[2],"% variance")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  labs(subtitle = "CA1")
print(pca12)

print(summary(aov(PC1 ~ treatment, data=pcadata)))
print(TukeyHSD(aov(PC1 ~ treatment, data=pcadata), which = "treatment"))

print(summary(aov(PC2 ~ treatment, data=pcadata))) 
print(TukeyHSD(aov(PC2 ~ treatment, data=pcadata), which = "treatment")) 

pca34 <- ggplot(pcadata, aes(PC3, PC4,color = treatment)) + 
  geom_point(size = 2, alpha = 1) +
  stat_ellipse(type = "t") +
  xlab(paste0("PC3: ", percentVar[3],"% variance")) +
  ylab(paste0("PC4: ", percentVar[4],"% variance")) +
  theme_cowplot(font_size = 8, line_size = 0.25) +
  labs(subtitle = "CA1")
print(pca34)

print(summary(aov(PC4 ~ treatment, data=pcadata))) 
print(TukeyHSD(aov(PC4 ~ treatment, data=pcadata), which = "treatment")) 


```


